tests/virtualenv_tools_test.py

```diff
@@ -61,6 +61,18 @@ def test_already_up_to_date(venv, capsys):
     assert out == 'Already up-to-date: {0} ({0})\n'.format(venv.before)
 
 
+def test_bourne_shell_exec(venv):
+    bindir = venv.before.join('bin')
+
+    bindir.join('bourne.py').write_binary(
+        bytes("""#!/bin/sh
+'''exec' {} "$0" "$@"
+' '''
+""".format(venv.before.strpath), 'utf-8'))
```

Comment:
this should use `write_text` and an fstring

unrelated because you'll get rid of it but: `bytes('...', 'utf-8')` is almost always wrong -- you want `'...'.encode()`

Reply:
Replaced with write(). I originally copied it from the non-ascii characters test and didn't think about why it was writing binary.